### ISSUE TITLE
TypeError: Cannot read property 'call' of undefined

### ISSUE THREAD

**Issue by kletteraffe11** [blainehansen/vue-async-properties#2]
Hi,
me again... thanks for fixing the last issue! Unfortunately I have another error:

`[Vue warn]: Error in created hook: "TypeError: Cannot read property 'call' of undefined"`

I'm using vue 2.3.4 and webpack 2.3.3. 

```javascript
import Vue from 'vue'
import App from './App'
import router from './router'
import axios from 'axios'
import VueAxios from 'vue-axios'
import VueAsyncProperties from 'vue-async-properties'

Vue.use(VueAxios, axios)
Vue.axios.defaults.baseURL = 'http://localhost:3000/'

Vue.use(VueAsyncProperties)
```

```javascript
 export default {
    asyncData: {
      switches() {
        return this.axios.get('/switches')
          .then(response => response.data)
      }
    }
  }
```

Thanks for the help!

**Comment by blainehansen** [comment]
I think I know what's going on here. Take a look at the transforms section of the docs:
https://github.com/blainehansen/vue-async-properties#transformation-functions

There's a default transform applied to all incoming promises, that returns the response data just like you're doing with your `then` function. What this means for you, is that that `response.data` transformation is happening *twice*, once by your promise chain which happens first, and then once by the plugin.

You have two options.

- Either remove your `then` and let the plugin handle that for you (if you still have problems after that then that's a different issue)
- Keep your then, and pass `{ transform: null }` when making the `Vue.use` call to prevent the plugin from applying the default transform.

**Comment by blainehansen** [comment]
Did that work? Any news?

**Comment by kletteraffe11** [comment]
Sorry for the late response!

I removed my `then` but it didn't change anything. Logging my variable at creation of the component returns `null`. Is there a way to debug this?

**Comment by blainehansen** [comment]
The variable should be null at the creation, so that's not the problem. 

You could try using a local copy of the package to debug. Here are some basic instructions, I'll work on putting more descriptive ones in the README soon.

The easiest way to work on this project locally is to put the package src files in your project:

- fork the package repository
- clone your fork of the repository to a directory outside of your project
- copy the files in the package's `src` to somewhere your project code can import it (maybe a directory like `src/plugins` in your project? it doesn't really matter)
- import from that directly: `import VueAsyncProperties from './src/plugins/async-properties' (or whatever place you choose to put things, just make sure you're importing the package's `index.js` file).
- once you've got it working, put your changed files back into your cloned version of the repository
- commit and push that, and file a pull request, and I'll incorporate the changes :)
- then you can delete the local version and just go back to importing it through npm

Sorry this is all the help I can give. I will work on fixing this as soon as I can, but I have other things to work on for the time being. Hopefully you'll be able to figure it out yourself! :)

**Comment by kletteraffe11** [comment]
Hi,
here is a little update for you. I just created a new projekt using [this](https://github.com/Belar/hapi-vue) boilerplate. Looking at the chrome developer tools the issue seems to be in the core.js:
```javascript
TypeError: Cannot read property 'call' of undefined
    at VueComponent.eval (core.js:47)
    at VueComponent.boundFn [as acall$refresh] (vue.common.js:185)
    at eval (async-data.js:48)
    at eval (lodash.js:4944)
    at baseForOwn (lodash.js:3001)
    at eval (lodash.js:4913)
    at forEach (lodash.js:9359)
    at VueComponent.created (async-data.js:43)
    at callHook (vue.common.js:2667)
    at VueComponent.Vue._init (vue.common.js:4228)
```
This is the function in chrome:
```javascript
function resolverForGivenFunction(propName, _ref, givenFunction, defaultValue, transformFunction, errorHandler) {
  var metaPending = _ref.metaPending,
      metaLoading = _ref.metaLoading,
      metaError = _ref.metaError;

  var _this = this;

  var assignValue = function assignValue(result) {

    if (!__webpack_require__.i(__WEBPACK_IMPORTED_MODULE_0_lodash__["isNil"])(result)) _this[propName] = result;else _this[propName] = defaultValue;
  };

  var assignPending = void 0;
  if (metaPending) {
    var pendingName = metaPending(propName);
    assignPending = function assignPending(val) {
      _this[pendingName] = val;
    };
  } else {
    assignPending = function assignPending(val) {};
  }

  var loadingName = metaLoading(propName);
  var assignLoading = function assignLoading(val) {
    _this[loadingName] = val;
  };

  var errorName = metaError(propName);
  var assignError = function assignError(val) {
    _this[errorName] = val;
  };

  return function () {
    var givenResult = givenFunction.call(_this);
    // This is line 47 

    if (!__webpack_require__.i(__WEBPACK_IMPORTED_MODULE_0_lodash__["isNil"])(givenResult) && typeof givenResult.then === 'function') {
      assignLoading(true);
      assignPending(false);
      assignError(null);

      givenResult.then(function (result) {
        assignValue(transformFunction(result));
      }).catch(function (error) {
        _this[propName] = null;
        assignError(error);
        errorHandler(error);

        assignValue(null);
      }).then(function () {
        assignLoading(false);
      });
    } else {
      assignError(null);
      assignValue(givenResult);
    }
  };
}
```
I don't understand all of the code yet and my time is limited as well. Hopefully I will find a solution till next week.

**Comment by kletteraffe11** [comment]
#3

**Comment by kletteraffe11** [comment]
Fixed the issue in #3
